### PR TITLE
#97 Check that a method is not bridged to prevent instance clash

### PR DESCRIPTION
**Pull request by mitch2na**
When running Katharsis that has AbstractResource class there are bridged methods that clash with actual implementation.

### CODE REVIEW

**Comment by ieugen** [comment]
+1, seems legit. travis seems to report a bad build but without error message.

**Comment by mitch2na** [comment]
Reactor Summary:
[INFO] 
[INFO] Katharsis parent POM ............................... SUCCESS [  1.186 s]
[INFO] katharsis-core ..................................... SUCCESS [  1.300 s]
[INFO] katharsis-rs ....................................... SUCCESS [  0.555 s]
[INFO] io.katharsis:katharsis-servlet ..................... SUCCESS [  0.079 s]
[INFO] katharsis-spring ................................... SUCCESS [  0.575 s]
[INFO] katharsis-examples ................................. SUCCESS [  0.129 s]
[INFO] Katharsis Dropwizard + MongoDB Example ............. SUCCESS [  1.313 s]
[INFO] Katharsis Jersey Example ........................... SUCCESS [  0.412 s]
[INFO] Katharsis Simple Dropwizard Example ................ SUCCESS [  0.134 s]
[INFO] Katharsis Wildfly Example .......................... SUCCESS [  0.336 s]
[INFO] Spring Boot Simple Example ......................... SUCCESS [  0.601 s]
[INFO] Katharsis build POM ................................ SUCCESS [01:33 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:49 min
[INFO] Finished at: 2016-08-24T20:11:32+00:00
[INFO] Final Memory: 72M/496M
[INFO] ------------------------------------------------------------------------
The command "./sonar.sh" exited with 0.

**Comment by mitch2na** [comment]
Build passed but sonar exited with 0

File: katharsis-core/src/main/java/io/katharsis/utils/ClassUtils.java [hunk 1]
```diff
@@ -17,11 +16,6 @@
  */
 public class ClassUtils {
```
Comment (meshuga) [inline katharsis-core/src/main/java/io/katharsis/utils/ClassUtils.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1118)

**Comment by meshuga** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request merged**